pkg/tkctl/cmd/check/check.go

```diff
@@ -0,0 +1,460 @@
+		Example: checkExample,
+		Run: func(_ *cobra.Command, _ []string) {
+			cmdutil.CheckErr(options.Run())
+		},
+	}
+	cmd.Flags().BoolVarP(&options.DryRun, "dry-run", "", true, "whether dry-run without cluster")
+	cmd.Flags().StringVarP(&options.Cloud, "cloud", "", "", "cloud provider")
+	cmd.Flags().StringVarP(&options.ValuesFile, "values", "f", "tidb-cluster.yaml", "tidb-cluster values file")
+	cmd.Flags().StringVarP(&options.Chart, "chart", "", "pingcap/tidb-cluster", "tidb cluster chart path, local path or remote repo path")
+	cmd.Flags().StringVarP(&options.ChartVersion, "chart-version", "", "v1.0.0", "tidb cluster chart version")
+	flags := cmd.PersistentFlags()
+	settings.AddFlags(flags)
+	settings.Init(flags)
+	return cmd
+}
+
+func (o *CheckOptions) Run() error {
+	var tidbcluster v1alpha1.TidbCluster
+	var tidbConfigMap corev1.ConfigMap
+	var tikvConfigMap corev1.ConfigMap
+	var pdConfigMap corev1.ConfigMap
+	var pumpConfigMap corev1.ConfigMap
+	var drainerConfigMap corev1.ConfigMap
+	var pumpSts *appsv1.StatefulSet
+	var drainerSts *appsv1.StatefulSet
+	var tidbSvc corev1.Service
+	out, err := o.renderChart()
+	if err != nil {
+		return err
+	}
+	for k, v := range out {
+		filename := filepath.Base(k)
+		switch filename {
+		case "tidb-cluster.yaml":
+			if err := yaml.Unmarshal([]byte(v), &tidbcluster); err != nil {
+				return err
+			}
+		case "tidb-configmap.yaml":
+			if err := yaml.Unmarshal([]byte(v), &tidbConfigMap); err != nil {
+				return err
+			}
+		case "tikv-configmap.yaml":
+			if err := yaml.Unmarshal([]byte(v), &tikvConfigMap); err != nil {
+				return err
+			}
+		case "pd-configmap.yaml":
+			if err := yaml.Unmarshal([]byte(v), &pdConfigMap); err != nil {
+				return err
+			}
+		case "pump-configmap-rollout.yaml":
+			if err := yaml.Unmarshal([]byte(v), &pumpConfigMap); err != nil {
+				return err
+			}
+		case "drainer-configmap-rollout.yaml":
+			if err := yaml.Unmarshal([]byte(v), &drainerConfigMap); err != nil {
+				return err
+			}
+		case "tidb-service.yaml":
+			if err := yaml.Unmarshal([]byte(v), &tidbSvc); err != nil {
+				return err
+			}
+		case "pump-statefulset.yaml":
+			if err := yaml.Unmarshal([]byte(v), &pumpSts); err != nil {
+				return err
+			}
+		case "drainer-statefulset.yaml":
+			if err := yaml.Unmarshal([]byte(v), &drainerSts); err != nil {
+				return err
+			}
+		}
+	}
+	if !componentVersionMatches(&tidbcluster, pumpSts, drainerSts) {
+		errorPrintln("component version mismatch")
+		return errors.New("cluster component version mismatch")
+	}
+	infoPrintln("component version matches, all versions are ", getImageVersion(tidbcluster.Spec.PD.Image))
+
+	o.checkTidbService(tidbSvc)
+	checkResourceSettings(&tidbcluster, pumpSts, drainerSts)
+
+	var tidbcfg *config_v2_1_16.TidbConfig
+	var pdcfg *config_v2_1_16.PdConfig
+	var tikvcfg *config_v2_1_16.TikvConfig
+	if cfg, exist := tidbConfigMap.Data["config-file"]; exist {
+		infoPrintln("#### TiDB Config ####")
+		fmt.Println(cfg)
```

Comment:
These configs can be quite long. It would be useful if we can control what will be printed via a verbose level or other flags.

Reply:
Good point! I'll try to use `verbose` flag.